server/src/main/java/org/elasticsearch/index/mapper/GeoShapeIndexer.java

```diff
@@ -51,7 +51,9 @@ public GeoShapeIndexer(Orientation orientation, String name) {
         if (geometry == null) {
             return Collections.emptyList();
         }
-        geometry = GeometryNormalizer.apply(orientation, geometry);
+        if (GeometryNormalizer.needsNormalize(orientation, geometry)) {
```

Comment:
Looking at the code, needsNormalize does quite an extensive check, so I presume this adds some performance cost?

Reply:
`needsNormalize` should run several times faster than the normalisation.